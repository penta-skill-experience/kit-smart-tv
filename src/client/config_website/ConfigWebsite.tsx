import Typography from '@mui/material/Typography';
import * as React from "react";
import Tabs from '@mui/material/Tabs';
import Tab from '@mui/material/Tab';
import Box from '@mui/material/Box';
import {PersonalizationPage} from "./PersonalizationPage";
import {LayoutPage} from "./LayoutPage";
import {AdminPage} from "./AdminPage";
import {useState} from "react";
import {SelectChangeEvent} from "@mui/material/Select";
import {LogInPage} from "./LogInPage";
import {Button, Grid} from "@mui/material";
import {AnnouncementsPage} from "./AnnouncementsPage";
import * as emailValidator from "email-validator";


interface TabPanelProps {
    children?: React.ReactNode;
    index: number;
    value: number;
}

function TabPanel(props: TabPanelProps) {
    const {children, value, index} = props;

    return (
        <div
            role="tabpanel"
            hidden={value !== index}
        >
            {value === index && (
                <Box sx={{ p: 3 }}>
                    <Typography>{children}</Typography>
                </Box>
            )}
        </div>
    );
}

export function ConfigWebsite() {
    //state variables and methods for login page
    const password = 'password123';
    const [logInInput, setLogInInput] = useState('');
    const [visible, setVisible] = useState(false);
    const [loggedInStatus, setLoggedInStatus] = useState(false);

    const handleInput = (
        event: React.ChangeEvent<HTMLInputElement>
    ) => {
        setLogInInput(event.target.value);
    };

    const handleLogIn= () => {
        // todo
        // connect with persistence
        if (logInInput === password) {
            setLoggedInStatus(true);
        } else {
            alert('password not correct')
        }
    };

    const handleClickShowPassword = () => {
        setVisible(!visible)
    }

    //state variables and methods for tabs
    const [pageNumber, setPageNumber] = React.useState(0);

    const handleChange = (event: React.SyntheticEvent, newValue: number) => {
        setPageNumber(newValue);
    };

    //state variables and methods for personalization page
    const [colorScheme,setColorScheme] = useState<string | null>(null);
    const [fontSize, setFontSize] = useState<string | null>(null);

    const handleColorSchemeChange = (
        event: React.MouseEvent<HTMLElement>,
        newColorScheme: string | null,
    ) => {
        setColorScheme(newColorScheme);
    };

    const handleFontSizeChange = (
        event: React.MouseEvent<HTMLElement>,
        newFontSize: string | null,
    ) => {
        setFontSize(newFontSize);
    };

    const [selectedLightImage, setSelectedLightImage] = React.useState('');

    const handleLightImageSelect = (event) => {
        setSelectedLightImage(event.target.value);
    };

    const [selectedDarkImage, setSelectedDarkImage] = React.useState('');

    const handleDarkImageSelect = (event) => {
        setSelectedDarkImage(event.target.value);
    };

    //state variables and methods for layout page
    const initialList = [];
    const [counter, setCounter] = useState(1);
    const [list, setList] = React.useState(initialList);
    const [widget, setWidget] = React.useState({
        id:0,
        name:'',
        position:'',
        configurable:false,
        colorSolid:false
    });

    const incrementCounter = () => setCounter(counter + 1);

    const handleWidgetSelection = (event: SelectChangeEvent) => {
        //todo
        //config is not always true
        const updatedValue = {
            id:counter,
            name:event.target.value,
            position:'',
            configurable:true,
            colorSolid:false
        }
        setWidget(updatedValue)
    };

    const handleAddWidget = () => {
        if (widget.name !== '') {
            const newWidget = {
                //todo
                //config is not always true
                id:counter,
                name:widget.name,
                position:'',
                configurable:true,
                colorSolid:false
            }
            setList(list.concat(newWidget));
            incrementCounter();
            console.log('Added widget ' + newWidget.name + ' with id ' + newWidget.id)
        }
    };

    const handleDeleteWidget = (id) => {
        console.log('Widget with id ' + id + ' is removed ')
        setList(list.filter(item => item.id !== id));
    }

    const handlePosition = (id, position) => {
        const newList = list.map((item) => {
            if (item.id === id) {
                const newWidget = { ...item, position: position }
                return newWidget;
            } else {
                return item;
            };
        });

        setList(newList);

    }

    const handleColorSolid = (id, isChecked) => {
        const newList = list.map((item) =>{
            if (item.id === id) {
                const newWidget = {...item, colorSolid: !isChecked}
                return newWidget;
            } else {
                return item;
            };
        });
        setList(newList)
    }

    //state variables and methods for admin password page
    const [newPassword, setNewPassword] = useState('')
    const [oldPassword, setOldPassword] = useState('')

    const handleOldPassword = (
        event: React.ChangeEvent<HTMLInputElement>
    ) => {
        setOldPassword(event.target.value);
    };

    const handleNewPassword = (
        event: React.ChangeEvent<HTMLInputElement>
    ) => {
        setNewPassword(event.target.value);
    };

    //todo
    //Connect with persistence
    const handlePasswordChange = () => {
        alert('Old Password is ' + oldPassword + ' New Password is ' + newPassword)
    };

    const handleChangeSave = () => {
        if (colorScheme === null || fontSize === null) {
            alert('Color scheme and font size must be chosen')
        }
    };

    //state variables and methods for announcements page
    const initialMailList = [];
    const [mailList, setMailList] = React.useState(initialMailList);
    const [verUser, setVerUser] = React.useState({
        mail:'',
        name:'',
    });

    const handleMailChange = (event: React.ChangeEvent<HTMLInputElement>) => {
        setVerUser({...verUser, mail:event.target.value});
    };

    const handleNameChange = (event: React.ChangeEvent<HTMLInputElement>) => {
        setVerUser({...verUser, name:event.target.value});
    };

    const handleAddMail = () => {
        const alreadyExists = mailList.some(item => verUser.mail === item.mail);
        if (!emailValidator.validate(verUser.mail)) {
            alert('E-Mail does not exist')
            return;
        }
        if (alreadyExists === true) {
            alert('This E-Mail already exists')
            return;;
        }
        if (verUser.name !== '' && verUser.mail !== '') {
            const newUser = {
                mail:verUser.mail,
                name:verUser.name,
            }
            setMailList(mailList.concat(newUser));
            return;
        }
        alert('Username and email have to be filled out')
    };

    const handleDeleteUser = (mail) => {
        setMailList(mailList.filter(item => item.mail !== mail));
    }

    //state variable for log out

    const handleLogout = () => {
        setLoggedInStatus(false);
    }

    function renderConfigWebsite() {
        if(loggedInStatus === false) {
            return (
                <LogInPage
                    logInInput={logInInput}
                    visible={visible}
                    handleInput={handleInput}
                    handleLogIn={handleLogIn}
                    handleClickShowPassword={handleClickShowPassword}
                >
                </LogInPage>
            );
        } else {
            return (
                <div>
                    <Box sx={{
                        border: 1,
                        backgroundColor:'text.primary',}
                    }>
                        <Grid container spacing={2} direction="row" alignItems="center">
                            <Grid item xs={2}></Grid>
                            <Grid item container xs={8} alignItems="center" justifyContent="center">
                                <Grid item container alignItems="center" justifyContent="center">
                                    <Grid item>
                                        <Typography variant="h6" color='white'>Admin Interface</Typography>
                                    </Grid>
                                </Grid>
                            </Grid>
                            <Grid item container xs={2} direction="column">
                                <Button onClick={handleLogout}>
                                    Log Out
                                </Button>
                            </Grid>
                        </Grid>
                    </Box>

                    <Box sx={{ borderBottom: 1, borderColor: 'divider' }}>
                        <Tabs value={pageNumber} onChange={handleChange}>
                            <Tab label="Personalization"/>
                            <Tab label="Layout"/>
                            <Tab label="Password"/>
                            <Tab label="Announcements"/>
                        </Tabs>
                    </Box>
                    <TabPanel value={pageNumber} index={0}>
                        <PersonalizationPage
                            colorScheme={colorScheme}
                            fontSize={fontSize}
                            handleColorSchemeChange={handleColorSchemeChange}
                            handleFontSizeChange={handleFontSizeChange}
                            selectedLightImage={selectedLightImage}
                            selectedDarkImage={selectedDarkImage}
                            handleLightImageSelect={handleLightImageSelect}
                            handleDarkImageSelect={handleDarkImageSelect}
                        >
                        </PersonalizationPage>

                    </TabPanel>
                    <TabPanel value={pageNumber} index={1}>
                        <LayoutPage
                            list={list}
                            widget={widget}
                            handleWidgetSelection={handleWidgetSelection}
                            handleAddWidget={handleAddWidget}
                            handleDeleteWidget={handleDeleteWidget}
                            handlePosition={handlePosition}
                            handleColorSolid={handleColorSolid}
                        >
                        </LayoutPage>
                    </TabPanel>
                    <TabPanel value={pageNumber} index={2}>
                        <AdminPage
                            oldPassword={oldPassword}
                            newPassword={newPassword}
                            handleOldPassword={handleOldPassword}
                            handleNewPassword={handleNewPassword}
                            handlePasswordChange={handlePasswordChange}
                            handleChangeSave={handleChangeSave}
                        >
                        </AdminPage>
                    </TabPanel>
                    <TabPanel value={pageNumber} index={3}>
                        <AnnouncementsPage
                            mailList={mailList}
                            verUser={verUser}
                            handleMailChange={handleMailChange}
                            handleNameChange={handleNameChange}
                            handleAddMail={handleAddMail}
                            handleDeleteUser={handleDeleteUser}
                        >
                        </AnnouncementsPage>
                    </TabPanel>
                </div>
            );
        }

    }

    return(<div>
            {renderConfigWebsite()}
        </div>
    );
}