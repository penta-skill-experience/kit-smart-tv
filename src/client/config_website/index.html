<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <title>Admin Interface</title>
    <meta content="width=device-width, initial-scale=1" name="viewport">
    <link
            href="https://fonts.googleapis.com/css?family=Roboto:300,400,500,700&display=swap"
            rel="stylesheet"
    />
    <style>
        #root {
            height: 100vh;
            width: 100vw;
        }
    </style>
</head>
<body>
<div id="root">
    <!-- react output will go inside here -->
</div>
</body>
</html>
