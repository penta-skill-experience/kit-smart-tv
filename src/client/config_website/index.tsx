import "./index.css";
import * as ReactDOM from "react-dom";
import * as React from "react";
import {WidgetLoader} from "../widget/WidgetLoader";
import {ConcreteWidgetConfigSaver} from "./ConcreteWidgetConfigSaver";

ReactDOM.render(
    // this is just an example to demonstrate how widgets are instantiated
    // todo: remove this.
    <div>
        <h1>Hello Admin!</h1>
        <div>
            {new WidgetLoader().getWidget("rss-feed").createConfigComponent({url: "www.example.com"}, new ConcreteWidgetConfigSaver())}
        </div>
    </div>,
    document.getElementById("root")
);