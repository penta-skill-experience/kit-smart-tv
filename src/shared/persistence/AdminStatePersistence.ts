import * as https from "https";
import config from "./persistence.config.json";

export class AdminStatePersistence {
    get refresh_token(): string {
        return this._refresh_token;
    }
    get access_token(): string {
        return this._access_token;
    }
    private _access_token: string;
    private _refresh_token: string;
    constructor() {
    }

    async login(password: string) : Promise<boolean> {
        const response = await fetch(`${config.DOMAIN}/api/sessions`, {
            method: 'POST',
            body: JSON.stringify(password), // string or object
            headers: {
                'Content-Type': 'application/json'
            }
        });
        try{
            const myJson = await response.json(); //extract JSON from the http response
            if(myJson){
                this._access_token = myJson.accessToken;
                this._refresh_token = myJson.refreshToken;
                return true;
            }
        }
        catch(e: any)
        {
            return false;
        }
        return false;
    }

    async logout(access_token: string, refresh_token: string): Promise<boolean> {
        const response = await fetch(`${config.DOMAIN}/api/sessions`, {
            method: 'DELETE',
            body: null, // string or object
            headers: {
                'Content-Type': 'application/json',
                'x-refresh': refresh_token,
                'Authorization': 'Bearer ' + access_token,
    }
        });
        try{
            const myJson = await response.json(); //extract JSON from the http response
            if(myJson){
                this._access_token = myJson.accessToken;
                this._refresh_token = myJson.refreshToken;
                return true;
            }
        }
        catch(e: any)
        {
            return false;
        }
        return false;
    }

    async getAdminLoginState(access_token: string, refresh_token: string): Promise<boolean> {
        const response = await fetch(`${config.DOMAIN}/api/sessions`, {
            method: 'GET',
            body: null,
            headers: {
                'Content-Type': 'application/json',
                'x-refresh': refresh_token,
                'Authorization': 'Bearer ' + access_token,
            }
        });
        try{
            const myJson = await response.json(); //extract JSON from the http response
            if(new Date(myJson.valid_until).valueOf() > Date.now()){
                return true;
            }
        }
        catch(e: any)
        {
            return false;
        }
        return false;
    }

    async setPassword(oldpw: string, newpw: string, access_token: string, refresh_token: string): Promise<boolean> {
        const response = await fetch(`${config.DOMAIN}/admin/update-password`, {
            method: 'PUT',
            body: JSON.stringify({password: oldpw, new_password: newpw}),
            headers: {
                'Content-Type': 'application/json',
                'x-refresh': refresh_token,
                'Authorization': 'Bearer ' + access_token,
            }
        });
        try { //extract JSON from the http response
            if ( "password updated" === JSON.stringify(response.body)) {
                return true;
            }
        } catch (e: any) {
            return false;
        }
        return false;
    }
}