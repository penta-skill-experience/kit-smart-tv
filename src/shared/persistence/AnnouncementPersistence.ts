import {Announcement} from "../../server/announcement_management/Announcement";
import {VerifiedUser} from "../values/VerifiedUser";

export class AnnouncementPersistence {

    setAnnouncements(announcements: Announcement[]) {
        //todo
    }

    getAnnouncements(): Announcement[] {
        //todo
        return [];
    }

    getVerifiedUsers(): VerifiedUser[] {
        //todo
        return [];
    }

    addVerifiedUser(verifiedUser: VerifiedUser) {
        //todo
    }

    removeVerifiedUser(verifiedUser: VerifiedUser) {
        //todo
    }
}