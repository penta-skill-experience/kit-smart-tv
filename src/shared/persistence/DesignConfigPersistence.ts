import {ConfigData} from "./data";
import config from "./persistence.config.json";

export class DesignConfigPersistence {

    getSelectedColorSchemeId(): Promise<string> {
        return this.getConfigData().then(configData => configData.colorScheme);
    }

    getSelectedBackground(): Promise<string> {
        return this.getConfigData().then(configData => configData.background);
    }

    getSelectedFontSizeId(): Promise<string> {
        return this.getConfigData().then(configData => configData.fontSize);
    }

    setSelectedColorSchemeId(id: string) {
        return this.getConfigData()
            .then(configData => {
                const configToSet = configData;
                configToSet.colorScheme = id;
                return this.setConfigData(configToSet)
            })
    }

    setSelectedBackground(id: string) {
        return this.getConfigData()
            .then(configData => {
                const configToSet = configData;
                configToSet.background = id;
                return this.setConfigData(configToSet)
            });
    }

    setSelectedFontSize(id: string) {
        return this.getConfigData()
            .then(configData => {
                const configToSet = configData;
                configToSet.fontSize = id;
                return this.setConfigData(configToSet)
            });
    }

    /* Queries the whole config object.
       Other getters cherry-pick from this object for utility.
     */
    getConfigData(): Promise<ConfigData> {
        return fetch(`${config.DOMAIN}/config`)
            .then((value: Response) => value.json());
    }

    setConfigData(configData : ConfigData) {

        const headers = new Headers();
        headers.append("x-refresh", sessionStorage.getItem('refreshToken'));
        headers.append("Authorization", `Bearer ${sessionStorage.getItem("accessToken")}`);
        headers.append("Content-Type", "application/json");

        const requestOptions = {
            method: 'PUT',
            headers: headers,
            body: JSON.stringify(configData),
        };

        return new Promise<void>((resolve, reject) => {
            fetch(`${config.DOMAIN}/config`, requestOptions)
                .then(response => {
                    const new_accessToken = response.headers.get('x-access-token');
                    if (new_accessToken) {
                        //if a new accessToken is provided, update it.
                        sessionStorage.setItem("accessToken", new_accessToken);
                    }

                    if (response.status === 200) {
                        resolve();
                    } else {
                        reject();
                    }
                })
                .catch(() => reject());
        })
    }
}