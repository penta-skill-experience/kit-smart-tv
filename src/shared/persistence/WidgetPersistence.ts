import {WidgetData} from "../../client/widget/WidgetData";
import config from "./persistence.config.json";
import {TokenHolderSingleton} from "./TokenHolderSingleton";

export class WidgetPersistence {

    setWidgetDataList(list: WidgetData[]): Promise<Response> {

        const headers = new Headers();
        headers.append("x-refresh", TokenHolderSingleton.instance.refreshToken);
        headers.append("Authorization", `Bearer ${TokenHolderSingleton.instance.accessToken}`);
        headers.append("Content-Type", "application/json");

        const body = {widgetDataList: list};

        const requestOptions = {
            method: 'PUT',
            headers: headers,
            body: JSON.stringify(body),
        };

        return fetch(`${config.DOMAIN}/widgets`, requestOptions);
    }

    getWidgetDataList(): Promise<WidgetData[]> {
        return fetch(`${config.DOMAIN}/widgets`)
            .then((value: Response) => value.json())
            .then(data => data.widgetDataList);
    }
}