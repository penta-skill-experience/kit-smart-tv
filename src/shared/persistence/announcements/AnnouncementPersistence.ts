import {VerifiedUser} from "../../values/VerifiedUser";
import {AnnouncementPersistenceFrontend} from "./AnnouncementPersistenceFrontend";
import {detectEnvironment, Environment} from "../../util/detectEnvironment";
import {AnnouncementPersistenceBackend} from "./AnnouncementPersistenceBackend";
import {Announcement} from "../../values/Announcement";

/**
 * Singleton that returns a frontend or backend implementation
 * depending on which system the JavaScript code is running on.
 */
export abstract class AnnouncementPersistence {

    // SINGLETON:

    private static readonly INSTANCE: AnnouncementPersistence = detectEnvironment() === Environment.NODE ?
        new AnnouncementPersistenceBackend() : new AnnouncementPersistenceFrontend();

    public static getInstance(): AnnouncementPersistence {
        return this.INSTANCE;
    }

    abstract setAnnouncements(announcements: Announcement[]): Promise<void>;

    abstract getAnnouncements(): Promise<Announcement[]>;

    abstract getVerifiedUsers(): Promise<VerifiedUser[]>;

    abstract setVerifiedUsers(users: VerifiedUser[]): Promise<void>;
}