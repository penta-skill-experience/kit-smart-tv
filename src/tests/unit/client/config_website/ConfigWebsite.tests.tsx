import 'jsdom-global/register';
import React from "react";
import {ConfigWebsite} from "../../../../client/config_website/ConfigWebsite";
import {shallow, configure, mount} from "enzyme";
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import toJson from "enzyme-to-json";
import {act} from 'react-dom/test-utils';
import {AdminStatePersistence} from "../../../../shared/persistence/AdminStatePersistence";
import {DesignConfigPersistence} from "../../../../shared/persistence/DesignConfigPersistence";
import {WidgetPersistence} from "../../../../shared/persistence/WidgetPersistence";
import {AnnouncementPersistence} from "../../../../shared/persistence/announcements/AnnouncementPersistence";
import {
    AnnouncementPersistenceFrontend
} from "../../../../shared/persistence/announcements/AnnouncementPersistenceFrontend";


configure({adapter: new Adapter()});

const getAdminStateMock = jest.spyOn(AdminStatePersistence.prototype, "login");

let wrapper;

describe("config website login page tests", () => {


    beforeEach(()=>{
        wrapper =  shallow(<ConfigWebsite initialLogInStatus={false}/>);
    });


   test("expect login page", async () => {
       expect(toJson(wrapper)).toMatchSnapshot();
   });

    test("test for password input", async () => {

        expect(wrapper.find("LogInPage").at(0).prop("logInInput")).toEqual("");
        act(() => {
            wrapper.find("LogInPage").prop("handleInput")({target: {value:"password123"}});
        })
        wrapper.update();
        expect(wrapper.find("LogInPage").at(0).prop("logInInput")).toBe("password123");
    });

    test("login test", () => {
        getAdminStateMock.mockImplementation(async () => {
            return new Promise<void>( resolve => resolve());
        });
        act(() => {
            wrapper.find("LogInPage").prop("handleLogIn")({target: {value: ''}});
        });
        wrapper.update();
        expect(getAdminStateMock).toHaveBeenCalled();
    });

    afterAll(() => {
        jest.restoreAllMocks();
    });

});


describe("config website admin interface tests", () => {

    beforeEach(()=>{
        wrapper =  shallow(<ConfigWebsite initialLogInStatus={true}/>);
    });

    test("expect admin interface", async () => {
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    test("handleLogout function test", () => {
        const handleLogoutMock = jest.spyOn(AdminStatePersistence.prototype, "logout");
        handleLogoutMock.mockImplementation(() => {
            return new Promise<void>(resolve => {});
        });
        wrapper.find("ForwardRef(Button)").simulate("click");
        expect(handleLogoutMock).toHaveBeenCalled();
    });

    test("handleColorSchemeChange function test", () => {
        const newWrapper = wrapper.find("PersonalizationPage");
        newWrapper.prop("handleColorSchemeChange")({target:{event: "",newColorScheme:"light" }});
        expect(newWrapper.prop("colorScheme")).toEqual(null);
    });

    test("handleFontSizeChange function test", () => {
        const newWrapper = wrapper.find("PersonalizationPage");
        newWrapper.prop("handleFontSizeChange")({target:{event: "",newFontSize:"small"}});
        expect(newWrapper.prop("fontSize")).toEqual(null);
    });

    test("handlePersonalizationChange function test", () => {
        const setConfigDataMock = jest.spyOn(DesignConfigPersistence.prototype, "setConfigData");
        setConfigDataMock.mockImplementation(() => {
            return new Promise<void>(resolve => {});
        });
        const newWrapper = wrapper.find("PersonalizationPage");
        newWrapper.prop("handlePersonalizationChange")();
        expect(setConfigDataMock).toHaveBeenCalled();
    });

    test("handleWidgetSelection function test", () => {
        const newWrapper = wrapper.find("LayoutPage");
        newWrapper.prop("handleWidgetSelection")({target:{value: "tram-schedule"}});
        expect(newWrapper.prop("widgetListElement")).toEqual({
            "id": 0,
            "position": "",
            "widget": null,
            "widgetData": null,
            "widgetNameText": "",
        })
    });

    test("handleAddWidget function test", () => {
        const newWrapper = wrapper.find("LayoutPage");
        newWrapper.prop("handleAddWidget")();
        expect(newWrapper.prop("list")).toEqual([]);
    });

    test("handleDeleteWidget function test", () => {
        const newWrapper = wrapper.find("LayoutPage");
        newWrapper.prop("handleDeleteWidget")({id: "tram-schedule"});
        expect(newWrapper.prop("list")).toEqual([]);
    });

    test("handlePosition function test", () => {
        const newWrapper = wrapper.find("LayoutPage");
        newWrapper.prop("handlePosition")({id: "tram-schedule", position:1});
        expect(newWrapper.prop("list")).toEqual([]);
    });

    test("handleRawConfigSave function test", () => {
        const newWrapper = wrapper.find("LayoutPage");
        newWrapper.prop("handleRawConfigSave")({id: "tram-schedule", rawConfig:undefined});
        expect(newWrapper.prop("list")).toEqual([]);
    });

    test("handleLayoutChange function test", () => {
        const setWidgetDataListMock = jest.spyOn(WidgetPersistence.prototype, "setWidgetDataList");
        setWidgetDataListMock.mockImplementation(() => {
            return new Promise<Response>(resolve => {});
        })
        const newWrapper = wrapper.find("LayoutPage");
        newWrapper.prop("handleLayoutChange")();
        expect(setWidgetDataListMock).toHaveBeenCalled();
    });

    test("handleOldPassword function test", () => {
        act(() =>{
            wrapper.find("AdminPage").at(0).prop("handleOldPassword")({target:{value: "test"}});
        });
        wrapper.update();
        expect(wrapper.find("AdminPage").at(0).prop("oldPassword")).toBe("test");
    });

    test("handleNewPassword function test", () => {
        act(() =>{
            wrapper.find("AdminPage").at(0).prop("handleNewPassword")({target:{value: "test"}});
        });
        wrapper.update();
        expect(wrapper.find("AdminPage").at(0).prop("newPassword")).toBe("test");
    });

    test("handlePasswordChange function test", () => {
        const handlePasswordChangeMock = jest.spyOn(AdminStatePersistence.prototype, "setPassword");
        handlePasswordChangeMock.mockImplementation(() => {
            return new Promise<void>(resolve => {});
        });
        wrapper.find("AdminPage").prop("handlePasswordChange")({target:{}})
        expect(handlePasswordChangeMock).toHaveBeenCalled();
    });

    test("handleMailChange function test", () => {
        act(() => {
            wrapper.find("AnnouncementsPage").at(0).prop("handleMailChange")({target:{value: "test"}});
        });
        wrapper.update();
        expect(wrapper.find("AnnouncementsPage").prop("verUser")).toEqual({
            mail:'test',
            name:'',
            verUser:null,
        });
    });

    test("handleNameChange function test", () => {
        act(() => {
            wrapper.find("AnnouncementsPage").at(0).prop("handleNameChange")({target:{value: "test"}});
        });
        wrapper.update();
        expect(wrapper.find("AnnouncementsPage").prop("verUser")).toEqual({
            mail:'',
            name:'test',
            verUser:null,
        });
    });

    test("handleAddMail function no input test ", () => {
        act(() => {
            wrapper.find("AnnouncementsPage").at(0).prop("handleAddMail")();
            wrapper.find("AnnouncementsPage").at(0).prop("handleAddMail")();
        });
        wrapper.update();
        expect(wrapper.find("AnnouncementsPage").prop("verUser")).toEqual({
            mail:'',
            name:'',
            verUser:null,
        });
        expect(wrapper.find("AnnouncementsPage").prop("mailList")).toEqual([])
    });

    test("handleAddMail function invalid email test ", () => {
        act(() => {
            wrapper.find("AnnouncementsPage").at(0).prop("handleMailChange")({target:{value: "test"}});
            wrapper.find("AnnouncementsPage").at(0).prop("handleNameChange")({target:{value: "test"}});
            wrapper.find("AnnouncementsPage").at(0).prop("handleAddMail")();
        });
        wrapper.update();
        expect(wrapper.find("AnnouncementsPage").prop("verUser")).toEqual({
            mail:'test',
            name:'test',
            verUser:null,
        });
        expect(wrapper.find("AnnouncementsPage").prop("mailList")).toEqual([]);
    });

    test("handleAddMail function success ", () => {
        act(() => {
            wrapper.find("AnnouncementsPage").at(0).prop("handleMailChange")({target: {value: "test@gmail.com"}});
            wrapper.find("AnnouncementsPage").at(0).prop("handleNameChange")({target: {value: "test"}});
            wrapper.find("AnnouncementsPage").at(0).prop("handleAddMail")();
        });
        wrapper.update();
        expect(wrapper.find("AnnouncementsPage").prop("verUser")).toEqual({
            mail: 'test@gmail.com',
            name: 'test',
            verUser: null,
        });
        expect(wrapper.find("AnnouncementsPage").prop("mailList")).toEqual([{
                mail: 'test@gmail.com',
                name: 'test',
                verUser: {_email: 'test@gmail.com', _name: 'test'},
            }]);

    });

    test("handleDeleteUser function test", () => {
        act(() => {
            wrapper.find("AnnouncementsPage").at(0).prop("handleMailChange")({target: {value: "test@gmail.com"}});
            wrapper.find("AnnouncementsPage").at(0).prop("handleNameChange")({target: {value: "test"}});
            wrapper.find("AnnouncementsPage").at(0).prop("handleAddMail")();
            wrapper.find("AnnouncementsPage").at(0).prop("handleDeleteUser")({
                    mail: 'test@gmail.com',
                    name: 'test',
                    verUser: {_email: 'test@gmail.com', _name: 'test'}
                });
        });
        wrapper.update();
        expect(wrapper.find("AnnouncementsPage").prop("mailList")).toEqual([]);
    });

    test("handleVerUserList function test", () => {
        const setVerifiedUsersMock = jest.spyOn(AnnouncementPersistence.getInstance(), "setVerifiedUsers");
        setVerifiedUsersMock.mockImplementation(() => {
            return new Promise<void>(resolve => {});
        });
        wrapper.find("AnnouncementsPage").prop("handleVerUserList")();
        expect(setVerifiedUsersMock).toHaveBeenCalled();
    });

    afterAll(() => {
        jest.restoreAllMocks();
    });



});
