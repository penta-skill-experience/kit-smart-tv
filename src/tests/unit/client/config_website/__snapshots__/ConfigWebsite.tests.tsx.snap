// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hanlde methods test handle old password text field 1`] = `
<LogInPage
  handleClickShowPassword={[Function]}
  handleInput={[Function]}
  handleLogIn={[Function]}
  logInInput=""
  visible={false}
/>
`;
